init.cmd

```diff
@@ -0,0 +1,55 @@
+@ECHO OFF

```

Comment:
Sorry, I'm gonna take back my Approve vote for a minute while we discuss this :-)

I don't think Windows-specific scripts should go in the root of the repo, at least not with such a generic name. What are your thoughts for how this will extend to Mac and Linux?

Reply:
I agree that we need to have a cross platform solution here, but this is a tricky problem due to what ships inbox on Windows and Unix systems. There's no scripting language that's common between them.

Batch files don't run on MacOS. Powershell ships with Windows and is available on MacOS, but requires an installation.
MacOS ships with bash, which is sort of an option on Windows if we want to take a dependency on a Git bash window (WSL can't meet our needs, we need a real Windows environment for everything other than bash). It also ships with Perl/Python, which requires an installation.

We might just have to have two scripts that do identical things that are implemented in different languages. Looking at projects similar to ours like the dotnet family of repos, they ship .cmd/.sh copies of each of their scripts. 

https://github.com/dotnet/corefx

Reply:
Yep, @nickgra, this is the direction I'm thinking we head. I don't want to hide important scripts down two levels of the tree. I'm happy to consider other names but I don't have a better suggestion than `init`ialize the environment.

Reply:
What's wrong with a `Scripts\Windows\init.cmd` and `Scripts/Mac/init.sh` approach? Again, these scripts are not (in my opinion) designed for discoverability by a new person, so I would follow our existing organizational pattern if possible.

Reply:
What's wrong with having them at the root? Why make me type "Scripts\Windows\init" instead of init? I think we should have these at the root for clarity, and we can squirrel the rest of implementation details in the Scripts\[Platform] pattern. 

I think init is special because it provides the facade over `Scripts\Windows\BuildGVFSForWindows` and `Scripts\Mac\BuildGVFSForMac`. A user, on either platform, types "init" and then "b". At least that part of the docs can be platform-agnostic, and it allows for us to make more of engineering system platform-agnostic in the future.

Reply:
I see, yea so just have an init.cmd and init.sh in the root? That can work, but two questions with that:
* Why does Windows get two scripts? Do we need both .cmd and .ps1?
* How do we differentiate Mac and Linux, since both support .sh?

Reply:
Unfortunately running the .bat from powershell doesn't persist the environment as the child process terminates, which is why we extract the environment into a temp file and read it into the powershell environment. I think we should provide one, but we could consider making it less in your face and moving it to Scripts\Windows

WRT sh support, my hope is we don't need to differentiate Mac and Linux at this layer. It's really just crafting the environment and providing some convenience shortcuts. If we needed to fork those shortcuts, I'd guess we can just interrogate `uname` and set the environment pointing at the platform we think we're running on. However, I haven't tested this yet.

Reply:
We will definitely need to fork some of the shortcuts. e.g. the Mac build scripts invoke xcodebuild, and Linux most definitely will not, so we will end up with Linux-specific build scripts.

On the filp side, some of the commands are already cross-plat-ready, e.g. invoking the unit tests just needs to execute a .NET Core application (though the scripts are currently doing slightly different things so they would need some work to be totally unified)